AsyncDisplayKit/ASCollectionView.mm

```diff
@@ -880,14 +880,15 @@ - (CGSize)viewportSizeForRangeController:(ASRangeController *)rangeController
 - (ASInterfaceState)interfaceStateForRangeController:(ASRangeController *)rangeController
 {
   ASCollectionNode *collectionNode = self.collectionNode;
-  if (collectionNode) {
+  if (collectionNode && [collectionNode supportsRangeManagedInterfaceState]) {
+    // Only use the interfaceStatate of nodes that are range managed
     return collectionNode.interfaceState;
-  } else {
```

Comment:
I prefer to keep the if / else rather than an early return.  It makes the code more clear in my opinion that there are exactly 2 cases that the code is handling.